ZipIterator.hpp

```diff
@@ -43,6 +44,11 @@ class ZipRef {
     if constexpr(I+1 < sizeof...(T)) swap_data<I+1>(o);
   }
 
+  template<std::size_t N = 0>
+  decltype(auto) get() {return *std::get<N>(ptr);}
+  template<std::size_t N = 0>
+  decltype(auto) get() const {return *std::get<N>(ptr);}
+
```

Comment:
I would prefer to have 
```
auto & 
const auto &
```
instead of `decltype(auto)`

Reply:
The problem is that constness in that case inherits from the constness of the ZipRef, not the constness of the referred-to points.

As-is, the example main.cpp compiles. However, if we replace `decltype(auto)`  with `auto &` and `const auto &`, then the example fails:

```
main.cpp: In function 'int main(int, char**)':
main.cpp:39:37: error: binding reference of type 'int&' to 'const int' discards qualifiers
   39 |   for(const auto &[x, y]: Zip(ai, bi)) {
      |                                     ^
main.cpp:39:37: error: binding reference of type 'std::__cxx11::basic_string<char>&' to 'const std::__cxx11::basic_string<char>' discards qualifiers
main.cpp:44:19: error: binding reference of type 'int&' to 'const int' discards qualifiers
   44 |     auto &[x,y] = z;
      |                   ^
main.cpp:44:19: error: binding reference of type 'std::__cxx11::basic_string<char>&' to 'const std::__cxx11::basic_string<char>' discards qualifiers
```

Maybe there's a better way to propagate constness of references?

As it is, I'm not yet able to provide `for(auto &[x, y]: zip(a, b))` because the ZipRef item needs to be a const ref, but x and y need to be mutable references.

Reply:
Please forgive my delay.
I'd like to better understand this, but so far I haven't been able to dedicate enough time.
I will do this ASAP.
In the meanwhile I wanted to let you know that your PR is much appreciated.